src/Swarm/Game/Scenario/Scoring/Progress.hs

```diff
@@ -0,0 +1,59 @@
+module Swarm.Game.Scenario.Scoring.Progress where
+
+import Data.Aeson
+import Data.Ord (Down (Down))
+import GHC.Generics (Generic)
+import Swarm.Util (maxOn)
+
+-- | This is a subset of the "ScenarioStatus" type
+-- that excludes the "NotStarted" case.
+data Progress
+  = Attempted
+  | Completed
+  deriving (Eq, Ord, Show, Read, Generic)
+
+untaggedJsonOptions :: Options
+untaggedJsonOptions =
+  defaultOptions
+    { sumEncoding = UntaggedValue
+    }
+
+instance FromJSON Progress where
+  parseJSON = genericParseJSON untaggedJsonOptions
+
+instance ToJSON Progress where
+  toJSON = genericToJSON untaggedJsonOptions
+
+data Metric a = Metric Progress a
+  deriving (Eq, Ord, Show, Read, Generic, FromJSON, ToJSON)
```

Comment:
I like the short modules, but it took me a while to find the `Metric` type hidden in `Progress` instead of say `Metrics`.

@kostmo please add module docs and export list. If you find it hard to write one because the functions and types don't make sense in a module with that name, maybe consider a different name or partition.

There may not be technical reasons for partitioning the module, but some humans (I hear) like to give meaning to such nonsense so their little brains can handle numerous modules more easily.

Reply:
Reorganized the modules.